src/internal/connector/onedrive/collections.go

```diff
@@ -325,6 +328,33 @@ func (c *Collections) Get(
 		folderPaths[driveID] = map[string]string{}
 		maps.Copy(folderPaths[driveID], paths)
 
+		// Set all folders in previous backup but not in the current
+		// one with DoNotMerge
+		if c.ctrl.ToggleFeatures.EnableOneDriveDeltaIncrementals && delta.Reset {
+			for i, p := range oldPaths {
+				_, found := paths[i]
+				if !found {
+					prevPath, err := path.FromDataLayerPath(p, false)
+					if err != nil {
```

Comment:
Nit: deep nesting always feels like a smell to me.  Some suggestions:
```suggestion
		if !c.ctrl.ToggleFeatures.EnableOneDriveDeltaIncrementals || !delta.Reset {
		    maps.Copy(excludedItems, excluded) // pulled up from 358
		    continue
		}
		for i, p := range oldPaths {
			_, found := paths[i]
			if found {
			    continue
			}
			
			prevPath, err := path.FromDataLayerPath(p, false)
			if err != nil {
```

Reply:
+1 for less nesting

Small typo in the suggestion above: it should only copy excluded items if incrementals were enabled and the delta wasn't reset (above has extra negation in there)

it could probably be turned into something like
```go
if !c.ctrl.ToggleFeatures.EnableOneDriveDeltaIncrementals {
  continue
}

if !delta.Reset {
  maps.Copy(exclucdedItems, excluded)
  continue
}

// code for finding/making collections for deleted folders
```